src/main/java/org/jvnet/hudson/test/TemporaryDirectoryAllocator.java

```diff
@@ -50,8 +50,15 @@ public TemporaryDirectoryAllocator(File base) {
     }
 
     public TemporaryDirectoryAllocator() {
-        this.base = new File(System.getProperty("java.io.tmpdir"), "jenkinsTests.tmp");
-        base.mkdirs();
+        File tempDir = new File(System.getProperty("java.io.tmpdir"), "jenkinsTests.tmp");
+        if (tempDir.mkdirs() == false) {
+            try {
+                File tempDir2 = java.nio.file.Files.createTempDirectory("jenkinsTests").toFile();
```

Comment:
Because createTempDirectory does not provide a fixed name, and I didn't want to risk impacting tests which inadvertently depended on a fixed name directory.  I assumed (possibly incorrectly) that some test might have wanted the fixed directory name, and if it is available and writable, then I didn't see any harm leaving it that way.

Windows users are typically single user on a box, so they don't mind having a fixed directory name in the temporary directory.

MacOS users are probably single user on a box, so they likely don't mind.